app/gui/src/project-view/components/CodeEditor/CodeEditorImpl.vue

```diff
@@ -17,58 +18,49 @@ import {
 } from '@codemirror/language'
 import { lintGutter } from '@codemirror/lint'
 import { highlightSelectionMatches } from '@codemirror/search'
-import { EditorState } from '@codemirror/state'
-import { EditorView } from '@codemirror/view'
 import { type Highlighter } from '@lezer/highlight'
 import { minimalSetup } from 'codemirror'
-import { computed, onMounted, ref, watch, type ComponentInstance } from 'vue'
+import { computed, onMounted, useTemplateRef, watch, type ComponentInstance } from 'vue'
 
 const projectStore = useProjectStore()
 const graphStore = useGraphStore()
 const suggestionDbStore = useSuggestionDbStore()
-const editorRoot = ref<ComponentInstance<typeof EditorRoot>>()
+const editorRoot = useTemplateRef<ComponentInstance<typeof CodeMirror>>('editorRoot')
 const rootElement = computed(() => editorRoot.value?.rootElement)
 useAutoBlur(rootElement)
 
-const editorView = new EditorView()
+const { editorView, setExtraExtensions } = useCodeMirror(editorRoot, {
+  extensions: [
+    minimalSetup,
+    syntaxHighlighting(defaultHighlightStyle as Highlighter),
+    bracketMatching(),
+    foldGutter(),
+    lintGutter(),
+    highlightSelectionMatches(),
+    ensoSyntax(),
+    ensoHoverTooltip(graphStore, suggestionDbStore),
+  ],
+})
 ;(window as any).__codeEditorApi = testSupport(editorView)
-
 const { updateListener, connectModuleListener } = useEnsoSourceSync(graphStore, editorView)
 const ensoDiagnostics = useEnsoDiagnostics(projectStore, graphStore, editorView)
-
 watch(
   () => projectStore.module,
   (module) => {
     if (!module) return
-    editorView.setState(
-      EditorState.create({
-        extensions: [
-          minimalSetup,
-          syntaxHighlighting(defaultHighlightStyle as Highlighter),
-          bracketMatching(),
-          foldGutter(),
-          lintGutter(),
-          highlightSelectionMatches(),
-          ensoSyntax(),
-          updateListener,
-          ensoHoverTooltip(graphStore, suggestionDbStore),
-          ensoDiagnostics,
-        ],
-      }),
-    )
+    setExtraExtensions([updateListener, ensoDiagnostics])
```

Comment:
Why these extensions need to be set in watch?

Reply:
They don't--moved them.